Added an article about tasks (#6718)
mesosphere/marathon: docs/articles/tasks.md

# Tasks

### What is a Task?

> Task is a process which is launched by Mesos executor on the Mesos Agent. When Marathon launches an instance of the application definition, Marathon sends a command to Mesos to launch one or more tasks.

Reviewer: > Task is a process which is launched by Mesos executor on the Mesos Agent

A new user can probably imagine waht "Mesos Agent" is but what is Mesos executor? Why is it needed? 

>  Marathon sends a command to Mesos to launch one or more tasks

I feel that this too simple for what actually happens under the hood. If a user is new to Mesos, this is not particularly helpful.

### How can I use a Task?

> Tasks are automatically created by Mesos once application definition is being launched.

Reviewer: > Tasks are automatically created by Mesos

What does "automatically" specifically means? It feels like the description above still doesn't quite explain what a task is. How about an example e.g .`"cmd": "sleep 100"`, describing the process of starting it on some agent (going through offer matching, executor etc.), throwing in a few points about isolation (with links to Mesos docs) and describing the fact, that user can see tasks sandbox (stderr, stdout)...

There are several other places where interaction with Tasks is possible. This includes:

##### API endpoints:
 * `/v2/apps/<app_id>/tasks`: Lists all running tasks for a provided application id, and gives you task specific information such as agent id, host, task state and so on. It’s also possible to kill tasks using this endpoint.
 * `/v2/tasks`: Lists all running tasks, and gives you task specific information such as agent id, host, task state and so on.

>  * `/v2/tasks/delete`: Endpoint for deleting one or more tasks

Reviewer: We tell the user, **what** Marathon have, but we don't say **why** to use this or that. When does the user should fetch `/v2/tasks`? When `/v2/apps/<app_id>/tasks`?  One can delete tasks via both - which one should the user take and why?

> ##### Marathon Health checks

Reviewer: What do Marathon Health checks have to do with `taskKillGracePeriodSeconds`??

> taskKillGracePeriodSeconds field allows you to set the amount of time between when the executor sends the SIGTERM message to gracefully terminate a task and when it kills it by sending SIGKILL.

Reviewer: ```suggestion
`taskKillGracePeriodSeconds` field allows you to set the amount of time between when the executor sends the SIGTERM message to gracefully terminate a task and when it kills it by sending SIGKILL.
```

##### Unreachable Strategy

> It’s possible to set the taks killing timeouts once a task enters unreachable state. See the Unreachable Strategy page for more information.

Reviewer: What do you mean? The unreachable strategy has nothing in it about _"killing timeouts"_

> It’s possible to set the taks killing timeouts once a task enters unreachable state. See the Unreachable Strategy page for more information.

Reviewer: ```suggestion
It’s possible to set the task killing timeouts once a task enters the unreachable state. See the Unreachable Strategy page for more information.
```

##### Environment Variables

> It’s possible to pass some environmental variables to the task. See the Environment Variables page for more information.

Reviewer: ```suggestion
It’s possible to provide the task with environmental variables. See the Environment Variables page for more information.
```

### I want to know more

> More detailed explanation of tasks can me found on [Apache Mesos website](http://mesos.apache.org/documentation/latest/).

Reviewer: ```suggestion
A more detailed explanation of tasks can be found on [Apache Mesos website](http://mesos.apache.org/documentation/latest/).
```

## How can I use a Task?

Tasks are automatically created by Mesos once a service definition is being launched.

There are several other places where interaction with Tasks is possible. This includes:

## API endpoints:

In Marathon, there are several API endpoints related to tasks. To learn more, check out the respective `apps`, `pods` and `tasks` sections on the [Marathon REST API page](http://mesosphere.github.io/marathon/api-console/index.html).

## Marathon Health Checks configuration

`taskKillGracePeriodSeconds` field allows you to set the amount of time between when the executor sends the `SIGTERM` message to gracefully terminate a task and when it kills it by sending `SIGKILL`. Please check the [health checks](health-checks.md) page for more information.

## Unreachable Strategy

There are a parameters to control the Marathon's treatment of unreachable tasks, with regards to how quickly a replacement task is launched, and how quickly Marathon treats the task as permanently gone. See the Unreachable Strategy page for more information.

## Environment Variables

> Tasks receive various environment variables, describing networking ports, Mesos metadata, and Marathon metadata. It’s also possible to specify custom environment variables. See the Environment Variables page for more information.

Reviewer 2: I suppose we don't have links here yet?
